File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/MobileBackdoorParams.java
package io.appium.espressoserver.lib.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

@SuppressWarnings("unused")
public class MobileBackdoorParams extends AppiumParams {
    private InvocationTarget target;
    private String elementId;

    private List<MobileBackdoorMethod> methods;

    public InvocationTarget getTarget() {
        return target;
    }

    public List<MobileBackdoorMethod> getMethods() {
        return methods;
    }

    public String getElementId() {
        return elementId;
    }

    public enum InvocationTarget {
        @SerializedName("activity")
        ACTIVITY,
        @SerializedName("application")
        APPLICATION,
        @SerializedName("element")
        ELEMENT,
    }

}

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/MobileBackdoorParams.kt
package io.appium.espressoserver.lib.model

import com.google.gson.annotations.SerializedName

data class MobileBackdoorParams(
        val target: InvocationTarget? = null,
        var targetElement: String? = null,
        val methods: List<MobileBackdoorMethod> = emptyList()

) : AppiumParams() {

    companion object {
        enum class InvocationTarget {
            @SerializedName("activity")
            ACTIVITY,
            @SerializedName("application")
            APPLICATION,
            @SerializedName("element")
            ELEMENT
        }
    }
}

